pandas/tools/merge.py

```diff
@@ -245,11 +248,15 @@ def _indicator_post_merge(self, result):
         result['_left_indicator'] = result['_left_indicator'].fillna(0)
         result['_right_indicator'] = result['_right_indicator'].fillna(0)
 
-        result[self.indicator_name] = Categorical((result['_left_indicator'] + result['_right_indicator']), categories=[1,2,3])
-        result[self.indicator_name] = result[self.indicator_name].cat.rename_categories(['left_only', 'right_only', 'both'])
-
-        result = result.drop(labels=['_left_indicator', '_right_indicator'], axis=1)
+        result[self.indicator_name] = Categorical((result['_left_indicator'] +
+                                                   result['_right_indicator']),
+                                                  categories=[1, 2, 3])
+        result[self.indicator_name] = (
+            result[self.indicator_name]
```

Comment:
I personally find this formatting very hard to read (though not real sure what's better)